### ISSUE TITLE
ConcurrentModificationException happened during delay exporting.

### ISSUE THREAD

**Issue by chickenlj** [apache/dubbo#9346]
```java
[03/12/21 14:30:09:436 CST] Dubbo-service-export-thread-1 ERROR config.ServiceConfig:  [DUBBO] Failed to export service config: org.apache.dubbo.demo.GreetingService, dubbo version: , current host: 30.225.11.234
java.util.ConcurrentModificationException
	at java.util.LinkedList$ListItr.checkForComodification(LinkedList.java:966)
	at java.util.LinkedList$ListItr.next(LinkedList.java:888)
	at org.apache.dubbo.common.config.CompositeConfiguration.getInternalProperty(CompositeConfiguration.java:76)
	at org.apache.dubbo.common.config.Configuration.getProperty(Configuration.java:124)
	at org.apache.dubbo.common.config.Configuration.getProperty(Configuration.java:111)
	at org.apache.dubbo.common.config.Configuration.convert(Configuration.java:144)
	at org.apache.dubbo.common.config.Configuration.getString(Configuration.java:48)
	at org.apache.dubbo.common.config.ConfigurationUtils.getDynamicProperty(ConfigurationUtils.java:147)
	at org.apache.dubbo.common.config.ConfigurationUtils.lambda$getCachedDynamicProperty$0(ConfigurationUtils.java:131)
	at org.apache.dubbo.common.config.ConfigurationCache.computeIfAbsent(ConfigurationCache.java:42)
	at org.apache.dubbo.common.config.ConfigurationUtils.getCachedDynamicProperty(ConfigurationUtils.java:131)
	at org.apache.dubbo.config.utils.ConfigValidationUtils.lambda$genCompatibleRegistries$0(ConfigValidationUtils.java:236)
	at java.util.ArrayList.forEach(ArrayList.java:1249)
	at org.apache.dubbo.config.utils.ConfigValidationUtils.genCompatibleRegistries(ConfigValidationUtils.java:231)
	at org.apache.dubbo.config.utils.ConfigValidationUtils.loadRegistries(ConfigValidationUtils.java:226)
	at org.apache.dubbo.config.ServiceConfig.doExportUrls(ServiceConfig.java:370)
	at org.apache.dubbo.config.ServiceConfig.doExport(ServiceConfig.java:353)
	at org.apache.dubbo.config.ServiceConfig.lambda$doDelayExport$0(ServiceConfig.java:235)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:266)
	at java.util.concurrent.FutureTask.run(FutureTask.java)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
[03/12/21 14:30:09:436 CST] Dubbo-service-export-thread-3 ERROR config.ServiceConfig:  [DUBBO] Failed to export service config: org.apache.dubbo.demo.TripleService, dubbo version: , current host: 30.225.11.234
java.util.ConcurrentModificationException
	at java.util.LinkedList$ListItr.checkForComodification(LinkedList.java:966)
	at java.util.LinkedList$ListItr.next(LinkedList.java:888)
	at org.apache.dubbo.common.config.CompositeConfiguration.getInternalProperty(CompositeConfiguration.java:76)
	at org.apache.dubbo.common.config.Configuration.getProperty(Configuration.java:124)
	at org.apache.dubbo.common.config.Configuration.getProperty(Configuration.java:111)
	at org.apache.dubbo.common.config.Configuration.convert(Configuration.java:144)
	at org.apache.dubbo.common.config.Configuration.getString(Configuration.java:48)
	at org.apache.dubbo.common.config.ConfigurationUtils.getDynamicProperty(ConfigurationUtils.java:147)
	at org.apache.dubbo.common.config.ConfigurationUtils.lambda$getCachedDynamicProperty$0(ConfigurationUtils.java:131)
	at org.apache.dubbo.common.config.ConfigurationCache.computeIfAbsent(ConfigurationCache.java:42)
	at org.apache.dubbo.common.config.ConfigurationUtils.getCachedDynamicProperty(ConfigurationUtils.java:131)
	at org.apache.dubbo.config.utils.ConfigValidationUtils.lambda$genCompatibleRegistries$0(ConfigValidationUtils.java:236)
	at java.util.ArrayList.forEach(ArrayList.java:1249)
	at org.apache.dubbo.config.utils.ConfigValidationUtils.genCompatibleRegistries(ConfigValidationUtils.java:231)
	at org.apache.dubbo.config.utils.ConfigValidationUtils.loadRegistries(ConfigValidationUtils.java:226)
	at org.apache.dubbo.config.ServiceConfig.doExportUrls(ServiceConfig.java:370)
	at org.apache.dubbo.config.ServiceConfig.doExport(ServiceConfig.java:353)
	at org.apache.dubbo.config.ServiceConfig.lambda$doDelayExport$0(ServiceConfig.java:235)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run$$$capture(FutureTask.java:266)
	at java.util.concurrent.FutureTask.run(FutureTask.java)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:748)
[03/12/21 14:30:25:905 CST] Dubbo-service-export-thread-2  WARN config.ServiceConfig:  [DUBBO] Use random available port(20880) for protocol dubbo, dubbo version: , current host: 30.225.11.234
```

**Comment by wangchengming666** [comment]
I'll fix it

**Comment by CrazyHZM** [comment]
Resolved in #9348
